jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/CompressExtension.java

```diff
@@ -71,6 +73,11 @@
     private static final int DECOMPRESS_BUF_SIZE = 8 * 1024;
     
     private final static boolean NOWRAP = true;
+    private final static int POOL_CAPACITY = -1;
```

Comment:
not sure if it is right to set this to infinite pool capacity

Reply:
With an infinite pool, what limits the size?  Threads? Connections?
What would you do with a smaller pool if more were needed?

Reply:
Nothing limits the size of the pool if this is set to -1, the pool size will just dynamically increase to the max number of objects you ever borrow from the pool simultaneously, if the pool size is set to a smaller size it will just immediately end the object on `release` if the pool is past capacity.

Reply:
Something always limits the size of the pool.... memory would be one thing.   If threads always take and return items, then threads will limit the size.  If items are associated with connections, then connections limit size etc.     Is there something that naturally limits how big the pool can grow?  What is the scenario in which the pool can grow forever?

Reply:
Okay well in this instance it looks like the `CompressExtension` is only using 1 `Deflater` and 1 `Inflater` for the life of the extension so the pool isn't doing much here. It would make more sense if the pool was shared, although it looks like this is the only usage of inflaters/deflaters in websocket.